relay-general/src/store/transactions/processor.rs

```diff
@@ -356,13 +356,22 @@ impl Processor for TransactionsProcessor<'_> {
             ) {
                 scrub_identifiers(&mut event.transaction)?;
                 if self.name_config.mark_scrubbed_as_sanitized {
+                    // TODO(iker): we also mark transaction as sanitized after
+                    // applying renaming rules. Once we remove this feature
+                    // flag, relay should only mark transactions as sanitized
+                    // once.
                     event
                         .transaction_info
                         .get_or_insert_with(Default::default)
                         .source
                         .set_value(Some(TransactionSource::Sanitized));
                 }
             }
+
+            self.apply_transaction_rename_rule(
```

Comment:
When `mark_scrubbed_as_sanitized` is enabled (which it will be soon), the rules matching on `source: url` will fail to match, so `apply_transaction_rename_rule` will essentially become dead code. Maybe the previously failing tests already reflected this, I did not check in detail.

This brings me back to the point that we should "mark as sanitized" after applying the rules.